From 4275a1e114de8601f31f2a6c61588223d03bc438 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Tue, 11 Dec 2012 17:47:20 -0800
Subject: [PATCH 012/793] char: xilinx_devcfg: Release lock in error path

xdevcfg_set_dap_en() aquired a spin lock but never released it
in case of an error.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Acked-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/char/xilinx_devcfg.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/char/xilinx_devcfg.c b/drivers/char/xilinx_devcfg.c
index 4b8b0b3..2e8a103 100644
--- a/drivers/char/xilinx_devcfg.c
+++ b/drivers/char/xilinx_devcfg.c
@@ -535,11 +535,15 @@ static ssize_t xdevcfg_set_dap_en(struct device *dev,
 
 	status = strict_strtoul(buf, 10, &mask_bit);
 
-	if (status)
+	if (status) {
+		spin_unlock_irqrestore(&(drvdata->lock), flags);
 		return status;
+	}
 
-	if (mask_bit > 7)
+	if (mask_bit > 7) {
+		spin_unlock_irqrestore(&(drvdata->lock), flags);
 		return -EINVAL;
+	}
 
 	xdevcfg_writereg(drvdata->base_address + XDCFG_CTRL_OFFSET,
 				(ctrl_reg_status |
-- 
1.7.5.4

